lib/es6/toolset.js

```diff
@@ -60,8 +60,14 @@ Toolset.prototype.initializePosix = function (install) {
         this.cCompilerPath = "gcc";
     }
 
+    if( this.options.generator ){
+        if (install) {
+            this.log.info("TOOL", "Using " + this.options.generator + " generator, as specified from commandline.");
+        }
+        this.generator = this.options.generator;
```

Comment:
It's not already set.  The code immediately after this point sets the generator to a computed default.
(Oh it COULD be set, but it might not be....)

Because initially I left the generator initialization as = null; but I modified that to initialize to the options.generator thinking I could later check if generator is set, don't set it.

Then there's the logging that's done to report what generator was used....